### PR TITLE
Grw-1889/integrate tier selector into price calculator

### PR DESCRIPTION
**Pull request by KarnellSchultz**
## Describe your changes

Integrate tier selector into price calculator.

- renamed the component to tierSelector to make it more general
- redid the story for the component to match the actual API.

**Not included**

Description text for each tier
Recommended item / text 

More PRs to come to update the design of the tier selector to match the most recent designs.

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

[GRW-1889]: https://hedvig.atlassian.net/browse/GRW-1889?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/store/src/components/TierSelector/TierSelector.tsx [hunk 1]
```diff
@@ -131,11 +134,71 @@ export const TierItem = ({
       <TitleContainer>
         <TitleItem>{title}</TitleItem>
         <TitleItem>
-          <SecondaryTextStyle>{price}</SecondaryTextStyle>
+          <SecondaryTextStyle>{price}/mån</SecondaryTextStyle>
```
Comment (robinandeer) [inline apps/store/src/components/TierSelector/TierSelector.tsx:137]: Didn't we have a helper/custom hook for this?

File: apps/store/src/components/TierSelector/TierSelector.tsx [hunk 2]
```diff
@@ -131,11 +134,71 @@ export const TierItem = ({
...
+  const handleClick = (id: string) => {
+    onValueChange?.(id)
+  }
+
+  if (offers.length === 1) {
+    return <input hidden readOnly name={FormElement.ProductOfferId} value={selectedOfferId} />
+  }
+
+  return (
+    <Root type="multiple">
```
Comment (robinandeer) [inline apps/store/src/components/TierSelector/TierSelector.tsx:164]: Is `Root` both exported and used here?

File: apps/store/src/components/TierSelector/TierSelector.tsx [hunk 3]
```diff
@@ -131,11 +134,71 @@ export const TierItem = ({
...
+        <HeaderWithTrigger>
+          {selectedOffer ? (
+            <>
+              <div>{selectedOffer.variant.typeOfContract}</div>
+              <SecondaryTextStyle>
+                {currencyFormatter.format(selectedOffer.price.amount)}/mån
+              </SecondaryTextStyle>
+            </>
+          ) : (
+            <div>Välj skydd</div>
```
Comment (robinandeer) [inline apps/store/src/components/TierSelector/TierSelector.tsx:182]: nit: translation :)

**Review by robinandeer — COMMENTED** [review]
Great work!

It just hit me that we are using an Accordion to represent the tier selector. Is that really what it is though? To me, it's more like a "dropdown menu" or a simple "collapsible".

Not asking you to change this PR but worth thinking about.

File: apps/store/src/components/TierSelector/TierSelector.tsx (hunk 1 above)
Comment (KarnellSchultz) [reply apps/store/src/components/TierSelector/TierSelector.tsx:137]: Don't know but I'll poke around and see what I can find.

**Review by robinandeer — APPROVED** [review]
Great work! Eager to try it out :D

Just had a small comment about passing correct props instead of manipulating data inside presentational React components. Feel free to address in another PR if you want.

File: apps/store/src/components/TierSelector/TierSelector.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,37 @@
+import { ComponentMeta, ComponentStory } from '@storybook/react'
+import { useState } from 'react'
+import { CurrencyCode } from '@/services/apollo/generated'
+import { TierSelector, TierSelectorProps } from './TierSelector'
+
+export default {
+  title: 'Car Tier Selector',
```
Comment (guilhermespopolin) [inline apps/store/src/components/TierSelector/TierSelector.stories.tsx:7]:
```suggestion
  title: 'Tier Selector',
```

File: apps/store/src/components/TierSelector/TierSelector.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  },
+  {
+    id: '2',
+    variant: { displayName: 'Helförsäkring 🍒', typeOfContract: 'Driving' },
+    price: { amount: 999, currencyCode: CurrencyCode.Sek, __typename: 'Money' },
+  },
+]
+
+const Template: ComponentStory<typeof TierSelector> = (props) => {
+  const [selected, setSelected] = useState(props.selectedOfferId)
```
Comment (guilhermespopolin) [inline apps/store/src/components/TierSelector/TierSelector.stories.tsx:30]:
```suggestion
  const [selected, setSelected] = useState(MockedOffers[0].id)
```

It's not possible to interact with the story at the moment.

image

**Review by guilhermespopolin — APPROVED** [review]

File: apps/store/src/components/TierSelector/TierSelector.tsx [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+
+export type TierSelectorProps = {
+  offers: Array<ProductOfferFragment>
+  selectedOfferId: string
+  onValueChange: Dispatch<SetStateAction<string>>
+}
+export const TierSelector = ({ offers, selectedOfferId, onValueChange }: TierSelectorProps) => {
+  const { t } = useTranslation()
+
+  const currencyFormatter = useCurrencyFormatter(offers[0].price.currencyCode)
```
[the remark this answers is not in the document]
Comment (robinandeer) [reply apps/store/src/components/TierSelector/TierSelector.tsx]: Hmm there's no default currency 😅 we made that mistake in the current solution -- Sweden is just another market ;)

If the component requires a currency to render we should be explicit about that. Not introduce fallbacks just to make TS happy

[the remark this answers is not in the document]
Comment (KarnellSchultz) [reply apps/store/src/components/TierSelector/TierSelector.tsx]: Tack tack - it's now a prop 👍

**Comment by KarnellSchultz** [comment]
Going to merge this now. Another PR is following this PR with style changes.

**Pull request merged**